api/test/integration/conftest.py

```diff
@@ -116,13 +119,20 @@ def build_and_up(env_mode: str, interval: int = 10, interval_build_env: int = 10
     os.makedirs(test_logs_path, exist_ok=True)
     with open(docker_log_path, mode='w') as f_docker:
         while values_build_env['retries'] < values_build_env['max_retries']:
-            if build:
+            if build and not build_managers_only:
                 current_process = subprocess.Popen(["docker", "compose", "--profile", env_mode,
                     "build", "--build-arg", f"WAZUH_BRANCH={current_branch}", 
                     "--build-arg", f"ENV_MODE={env_mode}",
                     "--no-cache"],
                     stdout=f_docker, stderr=subprocess.STDOUT, universal_newlines=True)
                 current_process.wait()
+            if build_managers_only:
```

Comment:
```suggestion
            if build and build_managers_only:
```

> Or split the condition above to avoid repetition

So no images are built when both `nobuild` and `build-managers-only` are used.